### ISSUE TITLE
Training implementation

### ISSUE THREAD

**Issue by cMancio00** [cMancio00/Super-Resolution#9]
Parameter of the methods should be the same as normal training. For parameter values refer to the [paper](https://arxiv.org/pdf/1707.02921)

- [x] Training must be a method of the class [SuperResolution](https://github.com/cMancio00/Super-Resolution/blob/development/SRM/network.py#L4)
- [x] Print average loss each epochs
- [x] Plot Training loss
- [x] Save the checkpoint
- [x] Start to implement a main method for training.

> [!CAUTION]
> For the final implementation the training must **NOT** be in the notebook

> [!NOTE]
> optimizer and loss must be passed
